holidays/countries/brunei.py

```diff
@@ -0,0 +1,447 @@
+from holidays.holiday_groups import ChineseCalendarHolidays, ChristianHolidays
+from holidays.holiday_groups import InternationalHolidays, IslamicHolidays
+
+
+class Brunei(
+    HolidayBase,
+    ChineseCalendarHolidays,
+    ChristianHolidays,
+    InternationalHolidays,
+    IslamicHolidays,
+):
+    """
+    A subclass of :py:class:`HolidayBase` representing public holidays
+    in Brunei Darussalam.
+
+
+    References:
+
+    - Based on: http://www.labour.gov.bn/Lists/Upcomming%20events/AllItems.aspx
+                http://www.labour.gov.bn/Download/GUIDE%20TO%20BRUNEI%20EMPLOYMENT%20LAWS%20-%20english%20version-3.pdf  # noqa: E501
+    - Checked with: https://asean.org/wp-content/uploads/2021/12/ASEAN-National-Holidays-2022.pdf  # noqa: E501
+                    https://asean.org/wp-content/uploads/2022/12/ASEAN-Public-Holidays-2023.pdf  # noqa: E501
+                    https://www.timeanddate.com/holidays/brunei/
+    - [Jubli Emas Sultan Hassanal Bolkiah]
+        https://www.brudirect.com/news.php?id=28316
+
+    Limitations:
+
+    - Brunei Darussalam holidays only works from 1984 onwards
+    - Islamic holidays
+
+
+    Country created by: `PPsyrius <https://github.com/PPsyrius>`__
+
+    Country maintained by: `PPsyrius <https://github.com/PPsyrius>`__
+    """
+
+    country = "BN"
+    default_language = "ms"
+    estimated_label = tr("%s* (*anggaran)")
+    supported_languages = ("en_US", "ms", "th")
+
+    # Special Cases.
+
+    golden_jubilee_hassanal_bolkiah = tr("Jubli Emas Sultan Hassanal Bolkiah")
+
+    special_holidays = {
+        2017: ((OCT, 5, golden_jubilee_hassanal_bolkiah),),
+    }
+
+    def __init__(self, *args, **kwargs):
+        ChineseCalendarHolidays.__init__(self)
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        IslamicHolidays.__init__(self, calendar=BruneiIslamicCalendar())
+
+        super().__init__(*args, **kwargs)
+
+    def _populate(self, year):
+        # Available post-Independence from 1984 afterwards
+        if year <= 1983:
+            return None
+
+        def _add_observed(dt: date) -> None:
+            """
+            If Public Holiday falls on either Friday or Sunday, in-lieu
+            observance are given out on the following Saturday or Monday.
+            """
+            if self.observed and (self._is_friday(dt) or self._is_sunday(dt)):
+                for name in self.get_list(dt):
+                    self._add_holiday(
+                        self.tr("%s - Diperhatikan") % name, dt + td(days=+1)
+                    )
+
+        super()._populate(year)
+
+        # New Year's Day.
+        # Awal Tahun Masihi
+        # Status: In-Use.
+
+        _add_observed(self._add_new_years_day(tr("Awal Tahun Masihi")))
+
+        # Lunar New Year
+        # Tahun Baru Cina
+        # Status: In-Use.
+
+        _add_observed(self._add_chinese_new_years_day(tr("Tahun Baru Cina")))
+
+        # Isra Mi'raj
```

Comment:
I recommend placing all Islamic holidays after Gregorian calendar holidays. This will prevent observed holidays from doubling. For example, now 2033-12-26 looks like `"Christmas Day (Observed); Eid al-Fitr (Observed); Eid al-Fitr* (*estimated) (Observed)"`.

Reply:
Fixed.